scripts/.common.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+# .common.sh by Dustyn Gibson <miigotu@gmail.com>
+
+#DEBUG=1
+
+BINUTILS="binutils-2.22"
+GCC="gcc-PS3"
+GDB="gdb-7.5.1"
+NEWLIB="newlib-1.20.0"
+
+TOOLCHAIN_ROOT=$(git rev-parse --show-toplevel)
+[ -z $TOOLCHAIN_ROOT ] && { echo 'Could not get the root of the toolchain. Please use git clone rather than a source archive.' && exit 1; }
+
+PS3DEV_SOURCES=${TOOLCHAIN_ROOT}/src
+PS3DEV_SCRIPTS=${TOOLCHAIN_ROOT}/scripts
+PS3DEV_DEPENDS=${TOOLCHAIN_ROOT}/depends
+
+PROC=$(nproc --all 2> /dev/null)
+([ -z ${PROC} ] || [ ${?} -eq 0 ] || [ ${PROC} -eq 0 ]) && PROC=4
+
+SUBMODULE_UPDATE="git submodule update --init --recursive"
+git ls-remote -h "$(git remote get-url origin)" > /dev/null
+[ $? -eq 0 ] && SUBMODULE_UPDATE=${SUBMODULE_UPDATE}" --remote"
```

Comment:
This is so if the ps3toolchain submodule references havent been updated on origin yet, but the submodules have new commits, we will get the updated depends no matter what. If we have net access it will check remote, otherwise local.

Reply:
But that's wrong isn't it?  If you don't do "git pull" you should not get anything updated, and if you _do_ "git pull" then you will get the submodules updated anyway because your workflow has already pushed a commit updating the pinned submodule commits...

Reply:
True, I guess we dont need the --remote at all, but we do need the init in case someone clones without git configured to automatically init submodules. In that case they would just have empty folders in src/

Reply:
Yes, the `--init` should be fine.  I'm still a bit hazy on how this will work with tarballs though.  :smile: